domains/client/domain-operator/src/domain_worker.rs

```diff
@@ -135,6 +207,16 @@ pub(super) async fn start_worker<
                 biased;
 
                 Some((slot, proof_of_time)) = new_slot_notification_stream.next() => {
+                    if let Some(ref snap_sync_orchestrator) = snap_sync_orchestrator {
+                        if !snap_sync_orchestrator.domain_snap_sync_finished(){
+                            debug!(
+                                "Domain snap sync: skipping bundle production on slot {slot}"
+                            );
+
+                            continue
+                        }
+                    }
```

Comment:
This (and the relayer error) is probably because the consensus chain sync oracle is used, when consensus chain sync finishes the domain chain sync just starts so the domain block is still missing.

We can use a customized domain sync oracle here and in other places (e.g. the relayer) which:
- When full sync is used, return the result of the consensus chain sync oracle
- When domain snap sync is used, return the result of `snap_sync_orchestrator.domain_snap_sync_finished`

Reply:
Nazar described a similar solution offline mentioning `SubspaceSyncOracle`

Reply:
>This error most likely arises when we try to access the inaccessible block and it makes sense because we don't have those blocks then.

Yeah, what I propose is to skip processing the slot notification/block import notification/relayer until the domain sync is finished (at that point the required domain block and state should be available). Using `snap_sync_orchestrator.domain_snap_sync_finished` also work in some places but IMO sync oracle is a better abstraction.

>Nazar described a similar solution offline mentioning SubspaceSyncOracle

Maybe the same thing, but we need a `DomainSyncOracle` which can't be used by the consensus chain.